pkg/work/work.go

```diff
@@ -0,0 +1,44 @@
+package work
+
+import (
+	"github.com/goharbor/harbor-scanner-clair/pkg/etc"
+	"sync"
+)
+
+// Worker must be implemented by types that want to use the worker pool.
+type Worker interface {
+	Task()
+}
+
+// Pool provides a pool of goroutines that can execute any Worker tasks
+// that are submitted
+type Pool struct {
+	tasks chan Worker
+	wg    sync.WaitGroup
+}
+
+func New(config etc.WorkPoolConfig) *Pool {
+	p := Pool{
+		tasks: make(chan Worker),
+	}
+	p.wg.Add(config.MaxGoroutines)
```

Comment:
It seems no call to `wg.Wait()`. Any thread is waiting for the Wg signal?

Reply:
Scheduling goroutines on demand, we no longer keep the pool.